### DISCUSSION TITLE
Is there a way to see the generated stylesheet styles nativewind produces?

### DISCUSSION THREAD

**Discussion by TowhidKashem** [nativewind/nativewind#1129] [Q&A] [answered]
I purchased a license for nativwindui just to get a few of the screens. I don't really want to install nativewind on my project just to support these so it would be cool if I could extract the stylesheet or inline styles that nativewind produces so I can use them without NW and all its dependencies in an existing project. Is there a way to see what these are?

**Comment by marklawlor** [comment]
NativeWind is both a compiler and a runtime. React Native doesn't support a `className` prop so a runtime is needed to process it. The runtime also allows for styles like `active:` to work, and include polyfills for other styles.  The generated styles also include information like specificity and are not "dropin" replacements for style objects.

**Reply by TowhidKashem** [reply]
At a very high level, how does the final nativewind -> react native step work? Does the final code that gets built and/or compiled by nativewind (and it's dependencies) get transformed into something akin to stylesheet values? I'm no expert on the matter but my gut says it needs to in order for the RN stylesheet engine to understand it. Even if it's not a one to one replacement, is there any way for me to get something that kind of looks like stylesheet values out of this? It would get me most of the way there and I could do the rest of the work by hand like active states and what not..

I'm happy to spend the time to manually sift through it and convert the values by hand. Like I said I just needed 2 specific screens (the ios chat flow) from nativewindui to use in my app. I would have liked to just kept the nativewind dependencies but version 4 just doesn't feel stable yet and it seems nativewindui doesn't support version 2:

https://nativewindui.com/installation/manual

**Reply by marklawlor** [reply]
The output format is internal and changes regularly between patch versions, so any work you do will only work for that single version. `nativewindui` uses many advanced features such as CSS variables, the generated styles will not be style objects. I think you underestimate the work involved and that you'll be basically rewriting NativeWind.

**Reply by TowhidKashem** [reply]
Sorry maybe I miscommunicated. I don't intend to create some kind of interop layer and expect that to keep working with future native wind versions. 

I just wanted to grab as much of the styles as I could as a one time thing and then just use these in regular RN stylesheets going forward. 

**Pseudo code:**

Nativewindui:

`<View className="w-6 items-center justify-center">...</View>`

Compiled code:

`<View style={{
  width: 60,
  justifyContent: 'center',
  alignItems: 'center'
}}>...</View>`

I go into some bundle somewhere in some temp directory and grab

```
{
  width: 60,
  justifyContent: 'center',
  alignItems: 'center'
}
```

now I can hard code it in my app like:

```
<View style={{
  width: 60,
  justifyContent: 'center',
  alignItems: 'center'
}}>
```

am happy 😌

Now maybe the above is wishful thinking and the compiled code looks nothing like the above and like [this](https://en.wikipedia.org/wiki/Brainfuck) instead. In that case I'm screwed (lol) Like I said I don't know how Nativewind works under the hood and I may be super naïve with the above thinking. 

I was just hoping there is some way to get something that kind of looks like stylesheet styles. If not, thats fine, just thought I'd ask before translating the whole ui by hand as that could have saved me some time..

**Reply by marklawlor** [reply, marked as answer]
If there was an easy solution I would have provided it.

The compiled code can look like that, but in this instance it does not. `nativewindui` uses to many advance styles which cannot be presented via a simple object.

I've started a discussion with some library authors about the possibility of using nativewind component libraries without nativewind installed in a project, but it has some big problems that need to be solved so I wouldn't expect to see a result anytime soon.
